akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala

```diff
@@ -75,35 +75,38 @@ class GracefulTerminationSpec
       ex.getMessage should include("Connection refused")
     }
 
-    "fail chunked response streams" in new TestSetup {
+    "fail chunked response streams" in {
+      // client system created on the outside, so we simulate that client is not shut down at the same time as server
       val clientSystem = ActorSystem("client")
-      val r1 =
-        Http()(clientSystem).singleRequest(nextRequest(), connectionContext = clientConnectionContext, settings = basePoolSettings)
-
-      // reply with an infinite entity stream
-      val chunks = Source
-        .fromIterator(() => Iterator.from(1).map(v => ChunkStreamPart(s"reply$v,")))
-        .throttle(1, 300.millis)
-      reply(_ => HttpResponse(entity = HttpEntity.Chunked(ContentTypes.`text/plain(UTF-8)`, chunks)))
-
-      // start reading the response
-      val response = r1.futureValue.entity.dataBytes
-        .via(Framing.delimiter(ByteString(","), 20))
-        .runWith(TestSink.probe[ByteString])
-      response.requestNext().utf8String should ===("reply1")
-
       try {
-        val termination = serverBinding.terminate(hardDeadline = 50.millis)
-        response.request(20)
-        // local testing shows the stream fails long after the 50 ms deadline
-        response.expectNext().utf8String should ===("reply2")
-        eventually {
-          response.expectEvent() shouldBe a[OnError]
+        StreamTestKit.assertAllStagesStopped {
+          new TestSetup {
+            val r1 =
+              Http()(clientSystem).singleRequest(nextRequest(), connectionContext = clientConnectionContext, settings = basePoolSettings)
+
+            // reply with an infinite entity stream
+            val chunks = Source
+              .fromIterator(() => Iterator.from(1).map(v => ChunkStreamPart(s"reply$v,")))
+              .throttle(1, 100.millis)
+            reply(_ => HttpResponse(entity = HttpEntity.Chunked(ContentTypes.`text/plain(UTF-8)`, chunks)))
+
+            // start reading the response
+            val responseEntity = r1.futureValue.entity.dataBytes
+              .via(Framing.delimiter(ByteString(","), 20))
+              .runWith(TestSink.probe[ByteString])(SystemMaterializer(clientSystem).materializer)
+            responseEntity.requestNext().utf8String should ===("reply1")
+
+            val termination = serverBinding.terminate(hardDeadline = 1.second)
```

Comment:
Basically, now the delay between calling `terminate` and the hardDeadline is basically broken if only the response entity hasn't been streamed out. This never worked correctly (#3209) but some delay was introduced but for other reasons that have now been taken out of the picture.